string/string_test.mbt

```diff
@@ -584,11 +584,11 @@ test "String::fold" {
   let s = "this is a long string"
   inspect!(s.fold(init=0, fn(acc, c) { acc + c.to_int() }), content="1980")
   inspect!(
-    s.fold(init=@list.Nil, @list.add) |> @list.rev(),
+    s.fold(init=@list.nil(), @list.add) |> @list.rev(),
```

Comment:
Can we support `@list.nil` (not `@list.nil()`) as generic value? /cc @Yu-zh @bobzhang

Reply:
We need to check whether the generic value is deeply immutable to avoid the problem with value restriction. But I think we can only support simple cases like `nil` here.